analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("stringParam"), LiveOut("stringParam"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("stringParam"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_TernaryOperatorReuseCaptures()
+    {
+        /*       Block 1
+         *     boolParameter
+         *        /   \
+         *       T     F
+         *      /       \
+         *   Block 2  Block 3
+         *   #0=st   #0="Hello"
+         *       \     /
+         *       Block 4
+         *        s=#0
+         *          |
+         *        Block5
+         *     boolParameter
+         *        /   \
+         *       T     F
+         *      /       \
+         *   Block 6  Block 7
+         *   #1=st2   #1="Hello"
+         *       \     /
+         *       Block 8
+         *        q=#1
+         *         |
+         *        Exit
+         */
+        const string code = """
+            string s = boolParameter ? st : "Hello";
+            string q = boolParameter ? st2 : "Hello";
+            """;
+        var context = CreateContextCS(code, additionalParameters: "string st, string st2");
+        context.ValidateEntry(LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st", "st2"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st", "st2"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st", "st2"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st", "st2"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("boolParameter", "st", "st2"), LiveOut("boolParameter", "st2"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("boolParameter", "st2"), LiveOut("st2"));
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("st2"), LiveOut("st2"));
+        context.Validate(context.Cfg.Blocks[7], null, LiveIn("st2"), LiveOut("st2"));
+        context.Validate(context.Cfg.Blocks[8], null, LiveIn("st2"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_NullCoalescingOperator_ConsequentCalls()
+    {
+        /*       Block 1
+         *       #0=s1
+         *     #0 is null
+         *        /   \
+         *       F     T
+         *       /     \
+         *    Block 2  Block 3 -+---F--> Block 4 --------+
+         *    #1=#0    #2=s2    |       #1=#2            |
+         *       \  #2 is null  |                        |
+         *        \     /       |                        |
+         *         Block 6 <-+  +--T--> Block 5--------->|
+         *          s=#1     |         #1="Hello"        |
+         *           |       +---------------------------+
```

Comment:
This is confusing. Branches should only go down. It will take more space, but it will be easier to read. You'll need some `|` on the left

I didn't get the Block 3->4 transition

Reply:
re-drawn 🖌️